service/src/main/java/bio/terra/externalcreds/config/ExternalCredsConfigInterface.java

```diff
@@ -43,4 +43,17 @@
   boolean getAuthorizationChangeEventsEnabled();
 
   Optional<String> getAuthorizationChangeEventTopicName();
+
+  @Value.Default
+  default boolean getEnableKmsEncryption() {
+    return false;
+  }
+
+  String getServiceGoogleProject();
+
+  Optional<String> getKeyRingId();
+
+  Optional<String> getKeyId();
+
+  Optional<String> getKeyRingLocation();
```

Comment:
I would have a nested config object called `KmsConfiguration` that contains non-optional fields for google project, key ring id, key id and location. Then in this object have an optional `KmsConfiguration` setting. Then you don't need the enable setting, you can be sure either all values are set or not, and the code can handle optionals more appropriately